### ISSUE TITLE
ModalBottomSheet usage of MediaQueryData.accessibleNavigation seems wrong

### ISSUE THREAD

**Issue by bleroux** [flutter/flutter#121098]
`MediaQueryData.accessibleNavigation` is used in:
- `_MaterialBannerState`: to remove the slide out animation. 
- `ScaffoldMessengerState`: to remove snackbar and banner exit animation, and to remove automatic hiding of the snackbar.
- `_SnackBarState`: to remove the fade out animation. 
- `_ModalBottomSheetState`: to disable the initial animation
https://github.com/flutter/flutter/blob/ae8d05184ef3d571349098daacaa2c7cf885b97b/packages/flutter/lib/src/material/bottom_sheet.dart#L565

MaterialBanner, Scaffold and SnackBar usage of accessibleNavigation are valid (letting the user choose when a visible element should be closed instead of auto closing this element).

ModalBottomSheet usage seems wrong because it cancels an initial animation and the comment suggests this was done as a workaround (which might be necessary at that time). 
```dart
        // Disable the initial animation when accessible navigation is on so
        // that the semantics are added to the tree at the correct time.
```

Removing ModalBottomSheet usage of accessibleNavigation would fix https://github.com/flutter/flutter/issues/119094.
